### PR TITLE
Quiz tests

### PR DESCRIPTION
**Pull request by fullcolorcoder**
Tests for quiz.js

Resolves: https://jira.corp.adobe.com/browse/MWPW-134984

**Test URLs:**
- Before: https://uar-integration--milo--adobecom.hlx.page/?martech=off
- After: https://quiz-tests--milo--fullcolorcoder.hlx.page/drafts/bradjohn/quiz-22/?martech=off

### CODE REVIEW

File: .github/gnav.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Description
```
Comment (JackySun9) [inline .github/gnav.md:1]: Why do we need this file?

File: test/blocks/quiz/mocks/quiz.html [hunk 2]
```diff
@@ -1,176 +1,25 @@
...
-        <div>
-          <div data-align="center" data-valign="middle">analytics-quiz</div>
-          <div data-align="center" data-valign="middle">UARv3</div>
-        </div>
-      </div>
+<div>
+  <div class="quiz">
+    <div>
+      <div data-align="center" data-valign="middle">quizUrl</div>
+      <div data-align="center" data-valign="middle"><a href="https://main--milo--adobecom.hlx.page/delta/app-recommender/">https://main--milo--adobecom.hlx.page/delta/app-recommender/</a></div>
```
Comment (echampio-at-adobe) [inline test/blocks/quiz/mocks/quiz.html:5]: Hi @fullcolorcoder. Take a look at Sabya's latest commit in his PR to see about handling test page URLs. Rares made a comment about not exposing real URLs in public test cases.

**Review by echampio-at-adobe — CHANGES_REQUESTED** [review]

File: libs/blocks/quiz/quiz.js [hunk 3]
```diff
@@ -10,21 +10,20 @@ import {
 } from './utils.js';
 import StepIndicator from './stepIndicator.js';
 
-async function loadFragments(fragmentURL) {
+export async function loadFragments(fragmentURL, createFragment) {
```
Comment (JackySun9) [inline libs/blocks/quiz/quiz.js:13]: you pass createFragment as parameter, I think it is not good as import it, and also you update here and not udpate the calling one line 285, that will cause issue

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: libs/blocks/quiz/quiz.js (hunk 3 above)
Comment (echampio-at-adobe) [reply libs/blocks/quiz/quiz.js:13]: I think you can see the issue that Jacky foresaw by looking at your code in the context of Elaine's data that loads the footer fragment. In the uar-integration branch, with her data, you can see "personalize your website experience" at the bottom of the content area when you navigate through the first question. 

https://uar-integration--milo--adobecom.hlx.page/drafts/borges/end-to-end/?q-category=photo

In your branch, though, the fragment url is still unresolved in the footer div. 

https://quiz-tests--milo--fullcolorcoder.hlx.page/drafts/borges/end-to-end/?q-category=photo

`<div class="quiz-footer"><a href="https://main--milo--adobecom.hlx.page/drafts/borges/end-to-end/fragments/footer-questionfour"></a></div>`

**Review by echampio-at-adobe — CHANGES_REQUESTED** [review]

File: .github/gnav.md (hunk 1 above)
Comment (narcis-radu) [reply .github/gnav.md:1]: We don't, it looks like the branch needs a sync - https://github.com/adobecom/milo/pull/1233

**Pull request closed without merging**